tests/integration/test_kubectl_delete.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_config_values
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+config = {"scrape_interval": "15s", "scrape_timeout": "10s"}
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_from_local_path(ops_test, charm_under_test):
+    """Deploy the charm-under-test."""
+    logger.debug("deploy local charm")
+
+    resources = {"unused-image": METADATA["resources"]["unused-image"]["upstream-source"]}
+    await ops_test.model.deploy(charm_under_test, application_name=app_name, resources=resources)
+
+    await ops_test.model.applications[app_name].set_config(config)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="blocked", timeout=1000)
```

Comment:
What happens if you remove the "timeout=1000" ? (same question for other similar lines below).

Reply:
The default timeout is too short for ghwf and therefore occasionally tests fail.

Reply:
I agree but I have `timeout=1000` _everywhere_.

Reply:
Good point. I feel your pain. Still please put a note somewhere in that file, highlighted in some way to make it clear these tests can fail for no apparent reason other than github having a flu. Perhaps we should file tickets on pytest operator or python libjuju issue trackers that their defaults are too short and/or having to set timeouts in their APIs does not lead to a robust testing infrastructure. I will be happy to create those tickets but please do put some note in our code.

Reply:
Or if you really have the time and want to be fancy this could be a use case for a decorator whose docstring makes clear it is being used to deal with the fact that tests can fail with default timeout so timeouts are being extended.